jinahub/indexers/SimpleIndexer/__init__.py

```diff
@@ -44,60 +44,30 @@ def __init__(
         self._flush = True
         self._docs_embeddings = None
 
-    @property
-    def index_embeddings(self):
-        if self._flush:
-            self._docs_embeddings = np.stack(self._docs.get_attributes('embedding'))
-            self._flush = False
-        return self._docs_embeddings
-
     @requests(on='/index')
-    def index(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
+    def index(self, docs: Optional['DocumentArray'] = None, parameters: Optional[Dict] = {}, **kwargs):
         """All Documents to the DocumentArray
         :param docs: the docs to add
         :param parameters: the parameters dictionary
         """
-        traversal_path = parameters.get('traversal_paths', self.default_traversal_paths)
-        flat_docs = docs.traverse_flat(traversal_path)
+        if not docs: return
+        traversal_paths = parameters.get('traversal_paths', self.default_traversal_paths)
+        flat_docs = docs.traverse_flat(traversal_paths)
         self._docs.extend(flat_docs)
```

Comment:
Does it work with `flat_docs` is empty?

Reply:
Addressed, please see [here](https://github.com/jina-ai/executors/pull/47/commits/f9fe67dfa7b6b8958a85e1dd5c5298b697fca959) for the change. Also added tests for this case, thank you.